frontend/src/app/main/ui/workspace/tokens/form.cljs

```diff
@@ -156,30 +156,43 @@
 
 (defonce form-token-cache-atom (atom nil))
 
-(mf/defc ramp
+(defn hex->value
+  [hex]
+  (when-let [tc (tinycolor/valid-color hex)]
+    (let [hex (str "#" (tinycolor/->hex tc))
+          [r g b] (c/hex->rgb hex)
+          [h s v] (c/hex->hsv hex)]
+      {:hex hex
+       :r r :g g :b b
+       :h h :s s :v v
+       :alpha 1})))
```

Comment:
Pure functions that do not access internal state should be put outside the component. This way they are created statically and not need to be create on each component instantiation.